### PR TITLE
Response rate analytics 

### PR DESCRIPTION
**Pull request by lwilliams717**
For clarity, the response rate is the percentage of users that have chatted with their match and received a response from at least one other user.

The data calculated includes all communities and not just one specific community. Therefore, there is no default community in the config file. 
Also, the data only includes the ESI and SDS community. The demo community was excluded from the calculation process since most of the demo community users are used for testing and not actual butterfly users chatting with one another.

Some of the data for the chat event does not contain a chat id, only a user id. That data was ignored in the calculation process.

### CODE REVIEW

**Review by arushirai1 — APPROVED** [review]
Great work so far, I really liked how well separated the code functionality was for the most part. It hits the main criteria, so I am merging. However, there are minor changes that need to be made and you can create an additional PR to address those requests. It would also be nice to add a unit test to make sure compute chat analytics works correctly. Please check out the rest of the comments. If we had more time, it would have been great to view the response rate by week and maybe made a quick visualization in Jupyter Notebook. It's been great working with you, Lauren!

File: pipeline/flows/analytics.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import pandas as pd
+import prefect
+from prefect import Flow, Parameter
+from prefect.tasks.core.constants import Constant
+from prefect.tasks.secrets import PrefectSecret
+
+from pipeline.extract.analytics import extract_analytics
+from pipeline.transform.analytics import (
+    filter_chat_community,
```
[the remark this answers is not in the document]
Comment (lwilliams717) [reply pipeline/flows/analytics.py:11]: okay, will do!

File: pipeline/types/raw_data.py [hunk 2]
```diff
@@ -15,3 +15,4 @@
 RawUserInterests = Dict[UserId, Dict[InterestCode, bool]]
 # Three-level dictionary that indicates which intents a user selected
 RawUserIntents = Dict[UserId, Dict[IntentCode, Dict[Side, bool]]]
+RawAnalytics = Dict
```
[the remark this answers is not in the document]
Comment (lwilliams717) [reply pipeline/types/raw_data.py:18]: got it, thanks!

File: pipeline/transform/chat_analytics.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        if chat_id in chating_users_match:
+            chating_users_match[chat_id].add(user_id)
+        else:
+            chating_users_match[chat_id] = {user_id}
+
+    # calculates the number of chats with multiple chatting users
+    chats_multiple_users = 0
+    for key in chating_users_match:
+        if len(chating_users_match[key]) > 1:
+            chats_multiple_users = chats_multiple_users + 1
```
[the remark this answers is not in the document]
Comment (lwilliams717) [reply pipeline/transform/chat_analytics.py:30]: Oooh, I thought that wouldn't work in python for some reason. I'll change it now lol

**Pull request merged**